### PR TITLE
[Android] fix focus event on android and add UI Tests

### LINKED ISSUE #5509: [Android] Missing Focused event on material entry
### Description

Entries with material design does noe emit the Focused event.

Tested on iOS and UWP to, only Android has problem.

### Steps to Reproduce

1. Add new Entry with Visual set to Material
2. Attach a Focused event handler
3. Let the entry get focus

### Expected Behavior
The event handler should fire

### Actual Behavior
The event hander does not fire

### Basic Information

- Version with issue: 3.6.0.220655
- Last known good version: n/a
- IDE: VS Pro 15.9.8
- Platform Target Frameworks: 
  - iOS:  n/a
  - Android: 9.0
  - UWP:  n/a
- Android Support Library Version: 28.0.0.1
- Nuget Packages:
- Affected Devices:

### Screenshots

N/A

### Reproduction Link

https://github.com/jenscski/MissingFocusedMaterialEntry

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
- propagate focus event on nested edit text with material up to parent
- Add automationid to TextCell so that I can use it during UI Test
- Add Material UI Test abilities to Core Gallery Pages

### Issues Resolved ### 
- fixes #5509 

### Platforms Affected ### 
- iOS
- Android

### Testing Procedure ###
Run Entry Control gallery and test focus pages

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls/CoreGallery.cs [hunk 1]
```diff
@@ -246,6 +246,7 @@ public CoreRootView()
 	[Preserve(AllMembers = true)]
 	internal class CorePageView : ListView
 	{
+		[Preserve(AllMembers = true)]
```
Comment (PureWeen) [inline Xamarin.Forms.Controls/CoreGallery.cs]: I had to add this otherwise *TitleAutomationId* was getting linked out

**Comment by PureWeen** [comment]
- ios10 UI test fail unrelated
- API 28 test failures unrelated
- everything else passed

File: Xamarin.Forms.Controls/ViewContainers/StateViewContainer.cs [hunk 2]
```diff
@@ -24,8 +24,11 @@ public StateViewContainer (Enum formsMember, T view) : base (formsMember, view)
 				BindingContext = view,
 				AutomationId = name + "StateLabel"
 			};
-			if (name != "Focus")
-				stateValueLabel.SetBinding (Label.TextProperty, name, converter: new GenericValueConverter (o => o.ToString()));
+
+			if(name == "Focus" || name == "Unfocused" || name == "Focus")
```
Comment (samhouts) [inline Xamarin.Forms.Controls/ViewContainers/StateViewContainer.cs]: Did you mean to put `name == "Focus"` twice?

```suggestion
			if (name == "Focus" || name == "Unfocused")
```

Comment (PureWeen) [reply Xamarin.Forms.Controls/ViewContainers/StateViewContainer.cs]: - Focused

**Pull request merged**